dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/OracleStatsConnector.java

```diff
@@ -30,19 +32,46 @@
 @ParametersAreNonnullByDefault
 public class OracleStatsConnector extends AbstractOracleConnector {
 
+  static final int MAX_DAYS = 10000;
+  static final Duration maxDuration = Duration.ofDays(MAX_DAYS);
+
   public OracleStatsConnector() {
     super(OracleConnectorScope.STATS);
   }
 
   @Override
   public void addTasksTo(List<? super Task<?>> out, ConnectorArguments arguments) throws Exception {
     StatsTaskListGenerator taskListGenerator = new StatsTaskListGenerator();
-    out.addAll(taskListGenerator.createTasks(arguments));
+    Duration queriedDuration = getQueriedDuration(arguments);
+    out.addAll(taskListGenerator.createTasks(arguments, queriedDuration));
   }
 
   @Nonnull
   @Override
   public String summary(String fileName) {
     return String.format("Oracle statistics saved to %s", fileName);
   }
+
+  static Duration getQueriedDuration(ConnectorArguments arguments) {
+    Duration queriedDuration = extractFromArgs(arguments);
+    if (queriedDuration.compareTo(Duration.ofDays(0)) == 1
+        && queriedDuration.compareTo(maxDuration) < 1) {
+      return queriedDuration;
+    } else {
+      throw new MetadataDumperUsageException(
+          String.format(
+              "The number of days must be positive and not greater than %s. Was: %s",
+              MAX_DAYS, queriedDuration.toDays()));
+    }
+  }
+
+  private static Duration extractFromArgs(ConnectorArguments arguments) {
+    int queriedDays;
+    if (arguments.getQueryLogDays() == null) {
+      queriedDays = 30;
```

Comment:
I suggest to define constant variable with default Duration. You can then simplify this code:

static final Duration defaultQueriedDaysDuration = Duration.ofDays(30);

...

if (arguments.getQueryLogDays() == null) {
    return defaultQueriedDaysDuration;
}
return return Duration.ofDays(arguments.getQueryLogDays())

Reply:
Done